main.py

```diff
@@ -109,54 +58,115 @@ def main(cfg: DictConfig):
         cfg.finetune.test_batch_size,
     )
 
+    logger.info("Calculating metrics on base model")
+    # TODO cache the base model metrics, and only recompute if the cache does not exist
     if not cfg.pretrained:
-        train(
-            model,
+        base_pre_loss = train(
+            base_model,
             device,
             pre_train_loader,
             cfg.pretrain.epochs,
             learning_rate=cfg.pretrain.lr,
         )
 
-    _, pre_acc = test(model, device, pre_test_loader)
+    # prepare the model for the unadapt method
+    ufm_model = copy.deepcopy(base_model)
 
-    # Finetune model
-    fine_model = copy.deepcopy(model)
-    model_fine_losses = train(
-        fine_model,
+    if cfg.metric == "accuracy":
+        _, base_pre_acc = metrics.calculate_loss_and_accuracy(
+            base_model, device, pre_test_loader
+        )
+    elif cfg.metric == "llm_tasks":
+        metrics.run_llm_benchmark(base_model, logger=logger, tag="base_model")
+
+    base_fine_losses = train(
+        base_model,
         device,
         fine_train_loader,
         cfg.finetune.epochs,
         learning_rate=cfg.finetune.lr,
     )
-    _, model_fine_acc = test(fine_model, device, fine_test_loader)
+    wandb.log({"base_fine_losses": base_fine_losses})
```

Comment:
Can you log stuff in categories using slashes? E.g. `"fine_tune/val_loss"`? Also, in practice, we'll instead probably log continuously within the fine-tuning function so we can monitor FT loss while it's going. I'd just remove this line.